### PR TITLE
java and genericjmx plugins

### PR DESCRIPTION
**Pull request by kitchen**
There's a new global param, which needs to be updated for the other platforms. I only have centos, fedora, debian, ubuntu, and freebsd available to me (main platform is ubuntu, others would be coming from DO boxes) and as far as I can tell genericjmx isn't available for redhat, and java isn't available at all for freebsd, so I'm not overly certain how that should be handled, if there should be a check in the class to see if it's a supported platform or...

Also, I'd love it if someone who is using the collectd java plugin and knows how it behaves when it comes to multiple `<Plugin "java">LoadPlugin "foo" ...` blocks, like if the JVMArg parameter is global across all java plugins, or just that block, or what. The only one I have a need for at the moment is genericjmx, so I tried to make everything generic to support that but there may be some undocumented behavior it does that I'm not aware of since I'm only using one java plugin.

Also, on ubuntu (12.04 anyways) this needs openjdk-6-headless to be installed because it needs that specific jvm's libjvm.so (because they don't have an alternative for libjvm.so, so the package is built against a specific jvm... silly :() so this isn't perfectly drop-in, but I tried :)

I'm all for making changes to this to get this to a mergable state. I am currently using it myself on my ubuntu 12.04 boxes, so it does in fact work, but there's probably a fair amount more work required to get it mergable :)

### CODE REVIEW

**Comment by kitchen** [comment]
weird, travis failed. I'll look at that in a bit.

File: manifests/params.pp [hunk 1]
```diff
@@ -19,6 +20,7 @@
       $service_name      = 'collectd'
       $config_file       = "${collectd_dir}/collectd.conf"
       $root_group        = 'root'
+      # FIXME: $java_dir
```
Comment (blkperl) [inline manifests/params.pp]: you can set it to undef for now

File: manifests/params.pp [hunk 2]
```diff
@@ -37,6 +40,7 @@
       $service_name      = 'collectd'
       $config_file       = '/etc/collectd.conf'
       $root_group        = 'root'
+      # FIXME: $java_dir
```
Comment (blkperl) [inline manifests/params.pp]: set to undef

File: manifests/params.pp [hunk 3]
```diff
@@ -46,6 +50,7 @@
       $service_name      = 'collectd'
       $config_file       = '/usr/local/etc/collectd.conf'
       $root_group        = 'wheel'
+      # FIXME: $java_dir
```
Comment (blkperl) [inline manifests/params.pp]: same as above

**Comment by blkperl** [comment]
Looks good, just need to fix the failing tests

**Comment by txaj** [comment]
I'm not using that plugin but form what I understand from the doc ( https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_java & https://collectd.org/documentation/manpages/collectd-java.5.shtml ) LoadPlugin instantiate the class with the above JVMArg of the block we're talking about. I believe there is a scoping for each <Plugin java> block

**Comment by kitchen** [comment]
If that's the case that's perfect. Maybe I'll turn collect::plugin::java into a define instead of an almost useless class?

-Jeremy

--
Insert PGP signature here.

**Comment by kitchen** [comment]
sorry I have been busy today and not able to work on this. busybusy! :(

**Comment by kitchen** [comment]
weird, the test failures look like they're coming from concat?

```
     [31m  Undefined variable "::id"; Undefined variable "id" at /home/travis/build/pdxcat/puppet-module-collectd/spec/fixtures/modules/concat/manifests/setup.pp:25 on node testing-worker-linux-docker-2d5c3702-3209-linux-5.prod.travis-ci.org[0m

```

**Comment by kitchen** [comment]
oh, I see the problem. missing facts. fixing.

**Comment by blkperl** [comment]
Please update the README.md with docs for the new plugins and then this is ready to merge

**Comment by blkperl** [comment]
Awesome!

**Pull request merged**